services/config_service.rb

```diff
@@ -28,6 +30,89 @@ def active_code_hosting_service_key(provider_credential: nil)
       return "#{provider_credential.provider_name}_#{self.ci_user.id}"
     end
 
+    def restart_any_pending_work
+      return unless Services.onboarding_service.correct_setup?
+
+      # this helps during debugging
+      # in the future we should allow this to be configurable behavior
+      return if ENV["FASTLANE_CI_SKIP_RESTARTING_PENDING_WORK"]
+
+      github_projects = projects(provider_credential: clone_user_provider_credential)
+      github_service = FastlaneCI::GitHubService.new(provider_credential: clone_user_provider_credential)
+
+      run_pending_github_builds(projects: github_projects, github_service: github_service)
+      enqueue_builds_for_open_github_prs_with_no_status(projects: github_projects, github_service: github_service)
+    end
+
+    # In the event of a server crash, we want to run pending builds on server
+    # initialization for all projects for the provider credentials
+    def run_pending_github_builds(projects: nil, github_service: nil)
+      logger.debug("Searching all projects for commits with pending status that need a new build")
+      # For each project, rerun all builds with the status of "pending"
+      projects.each do |project|
+        pending_build_shas_needing_rebuilds = Services.build_service.pending_build_shas_needing_rebuilds(project: project)
+        logger.debug("No pending work to reschedule for #{project.project_name}") if pending_build_shas_needing_rebuilds.count == 0
+
+        # Enqueue each pending build rerun in an asynchronous task queue
+        pending_build_shas_needing_rebuilds.each do |sha|
+          logger.debug("Found sha #{sha} that needs a rebuild for #{project.project_name}")
+          build_runner = FastlaneBuildRunner.new(
+            project: project,
+            sha: sha,
+            github_service: github_service,
+            work_queue: FastlaneCI::GitRepo.git_action_queue # using the git repo queue because of https://github.com/ruby-git/ruby-git/issues/355
+          )
+          build_runner.setup(parameters: nil)
+          Services.build_runner_service.add_build_runner(build_runner: build_runner)
```

Comment:
So, in general, services shouldn't really know about each other. Ideally we'd have an instance of something that uses these two two services: `config_service` and `build_runner_service`
It might need something like a `SystemStartupCoordinator` to act like a controller without a view (coordinators are controllers without views? I just made that up.)

Reply:
Yeah, I agree with you 💯%. We have a lot of services which already have other knowledge of each other - do you think this issue is okay to tackle in another PR where we introduce the `SystemStartupCoordinator` you mentioned - or would you like that done in this PR?